Corona-Warn-App/src/main/java/de/rki/coronawarnapp/util/ConnectivityHelper.kt

```diff
@@ -74,11 +76,17 @@ object ConnectivityHelper {
      * @return current network status
      *
      */
+    @SuppressLint("NewApi")
     fun isNetworkEnabled(context: Context): Boolean {
         val manager = context.getSystemService(Context.CONNECTIVITY_SERVICE) as ConnectivityManager
-        val activeNetwork: Network? = manager.activeNetwork
-        val caps: NetworkCapabilities? = manager.getNetworkCapabilities(activeNetwork)
-        return caps?.hasCapability(NetworkCapabilities.NET_CAPABILITY_VALIDATED) ?: false
+        return if (BuildVersionWrap.hasAPILevel(Build.VERSION_CODES.M)) {
+            val activeNetwork = manager.activeNetwork
+            val caps: NetworkCapabilities? = manager.getNetworkCapabilities(activeNetwork)
+            caps?.hasCapability(NetworkCapabilities.NET_CAPABILITY_VALIDATED) ?: false
+        } else {
+            val activeNetworkInfo = manager.activeNetworkInfo
+            activeNetworkInfo != null && activeNetworkInfo.isConnected
+        }
```

Comment:
maybe we could polish it a bit and make it more readable ? as follows:
```kotlin
fun isNetworkEnabled(context: Context): Boolean {
        val manager = context.getSystemService<ConnectivityManager>() ?: return false
        return when {
            Build.VERSION.SDK_INT >= Build.VERSION_CODES.M -> {
                val activeNetwork = manager.activeNetwork
                val caps: NetworkCapabilities? = manager.getNetworkCapabilities(activeNetwork)
                caps?.hasCapability(NetworkCapabilities.NET_CAPABILITY_VALIDATED) ?: false
            }
            else -> {
                val activeNetworkInfo = manager.activeNetworkInfo
                activeNetworkInfo != null && activeNetworkInfo.isConnected
            }
        }
   }
```
and not using the version wrapper in this case , then we don't have to annotate it with `SuppessLint`